src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Base64Decoder.cs

```diff
@@ -34,15 +34,102 @@ public static partial class Base64
         /// - InvalidData - if the input contains bytes outside of the expected base64 range, or if it contains invalid/more than two padding characters,
         ///   or if the input is incomplete (i.e. not a multiple of 4) and <paramref name="isFinalBlock"/> is <see langword="true"/>.
         /// </returns>
-        public static unsafe OperationStatus DecodeFromUtf8(ReadOnlySpan<byte> utf8, Span<byte> bytes, out int bytesConsumed, out int bytesWritten, bool isFinalBlock = true)
+        public static OperationStatus DecodeFromUtf8(ReadOnlySpan<byte> utf8, Span<byte> bytes, out int bytesConsumed, out int bytesWritten, bool isFinalBlock = true)
         {
-            if (utf8.IsEmpty)
+            OperationStatus status = OperationStatus.Done;
+            bytesConsumed = 0;
+            bytesWritten = 0;
+
+            while (utf8.Length > 0)
             {
-                bytesConsumed = 0;
-                bytesWritten = 0;
-                return OperationStatus.Done;
+                status = DecodeFromUtf8Core(utf8, bytes, out int localConsumed, out int localWritten, isFinalBlock);
+                bytesConsumed += localConsumed;
+                bytesWritten += localWritten;
+
+                if (status is not OperationStatus.InvalidData or OperationStatus.DestinationTooSmall)
+                {
+                    break;
+                }
+
+                utf8 = utf8.Slice(localConsumed);
+                bytes = bytes.Slice(localWritten);
+
+                if (!TrySkipWhitespace(utf8, out localConsumed))
+                {
+                    if (localConsumed > 0)
+                    {
+                        bytesConsumed += localConsumed;
+                        status = OperationStatus.Done;
+                    }
+
+                    // The end of the input has been reached.
+                    break;
+                }
+
+                if (localConsumed == 0)
+                {
+                    // First char isn't whitespace, but we didn't consume anything,
+                    // thus the input may have whitespace anywhere in between. So fall back to block-wise decoding.
+                    status = DecodeWithWhitespaceBlockwise(utf8, bytes, out bytesConsumed, out bytesWritten, isFinalBlock);
+                    break;
```

Comment:
Is the `break` here "better" than 
```suggestion
                    return DecodeWithWhitespaceBlockwise(utf8, bytes, out bytesConsumed, out bytesWritten, isFinalBlock);
```
?
If codegen is the same, then I'd prefer `return` as it's easiert to read / follow (exit method here).